test/es6/ES6PromiseAsync.js

```diff
@@ -1400,7 +1400,167 @@ var tests = [
+                            if (this.i > 2)
+                            {
+                                throw new TypeError('failure inside iterator');
+                            }
+
+                            this.i++;
+
+                            return {
+                                done: this.i == 5,
+                                value: getAsyncResolvePromise('Test #' + index + ' - ', 'p' + this.i)
+                            };
+                        }
+                    };
+                }
+            };
+
+            Promise.any(objectWithIterator).then(
+                v => echo(`Test #${index} - Success - ${JSON.stringify(v)}`),
+                e => echo(`Test #${index} - Failed - ${JSON.stringify(e)}`));
+        }
+    },
+    {
+        name: "Promise.any resolve and reject functions cannot be called multiple times",
+        body: function(index) {
+            let oldThen = Promise.prototype.then;
+
+            try {
+                var resolveFunc;
+                var rejectFunc;
+
+                Promise.prototype.then = function(resolve, reject) {
+                    console.log(`Test #${index} - Temp then handler called from Promise.any`);
+
+                    // Stash resolve functions
+                    resolveFunc = resolve;
+                    rejectFunc = reject;
+                    // Reset Promise#then
+                    Promise.prototype.then = oldThen;
+                    // Call builtin Promise#then
+                    Promise.prototype.then.call(this, resolve, reject);
+                }
+
+                Promise.any([getAsyncResolvePromise('Test #' + index + ' - ', 'p1')]).then(
+                    v => {
+                        console.log(`Test #${index} - Success: ${JSON.stringify(v)}`);
+                        // Call the stashed resolve function a second time.
+                        // It is a wrapper (defined in getAsyncResolvePromise) and will log something
+                        // but it does not trigger allSettled to fire.
+                        resolveFunc('p2');
+                        rejectFunc('e2');
+                    },
+                    e => console.log(`Test #${index} - Error: ${e}`)
+                );
+            } finally {
+                // Reset Promise#then in case of failure
+                Promise.prototype.then = oldThen;
+            }
+        }
+    },
+    {
+		name: "Promise.any called with empty iterator, calls Promise.resolve synchronously and passes abrupt completion to reject handler",
+		body: function (index) {
+			function FakePromise(fn) {
+			  function resolve() { echo(`Test #${index} - resolve called`); throw new Error('oops'); }
+			  function reject(e) { echo(`Test #${index} - reject called: ${e.message}`) }
+			  fn(resolve, reject);
+			  this.then = function(onResolve, onReject) {};
+			}
+
+			FakePromise.resolve = function() {};
+			Promise.any.call(FakePromise, []);

```

Comment:
This test case appears to be broken, or at least does not produce the result in the baseline.